pkg/sources/filesystem/filesystem.go

```diff
@@ -151,7 +151,13 @@ func (s *Source) scanFile(ctx context.Context, path string, chunksChan chan *sou
 	defer inputFile.Close()
 	logger.V(3).Info("scanning file")
 
-	reReader, err := diskbufferreader.New(inputFile)
+	reader, err := handlers.HandleSpecializedArchives(ctx, path, inputFile)
```

Comment:
Special archive handling in standard archive handling would allow us to get this functionality into all sources that currently support archive handling.

Reply:
I think I found a pretty decent way to make this work. Let me know what you think.